File: 1-TwoSum/1-TwoSum.java
// Last updated: 7/2/2025, 11:55:47 PM
class Solution {
    public int[] twoSum(int[] nums, int target) {
      Map<Integer , Integer> map = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            int rem = target- nums[i];
            if(map.keySet().contains(rem)) return new int[]{i , map.get(rem)};
            map.put(nums[i] , i);
        }
        return new int[0];
    }
}

File: 1-TwoSum/1-TwoSum.cpp
// Last updated: 7/2/2025, 11:55:46 PM
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mp;

        for(int i=0;i<nums.size();i++)
        {
            if(mp.find(target - nums[i]) != mp.end()) return {mp[target - nums[i]] , i};
            mp[nums[i]] = i;
        }

        return {};
    }
};